setup_nightly.py

```diff
@@ -39,10 +39,22 @@ def readme():
         "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Topic :: Scientific/Engineering :: Artificial Intelligence",
+        "License :: OSI Approved :: MIT License",
+        "Operating System :: OS Independent",
     ],
     packages=find_packages(include=["pycaret*"]),
     include_package_data=True,
     install_requires=required,
-    extras_require={"full": optional_required,
-                    "test": test_required + optional_required},
+    extras_require={
+        "analysis": required_optional[1:9],
+        "models": required_optional[11:15],
+        "tuners": required_optional[17:22],
+        "mlops": required_optional[24:30],
+        "nlp": required_optional[32:38],
+        "full": required_optional,
+    },
```

Comment:
I think it would make more sense to split them into separate requirements files, or just define them directly in setup.py. This will break the moment we add or remove a dependency.

Reply:
thought about it, but having 8 separate requirements files is unclear an unnecessary. Let's think if there's a way to keep the two files but make the code so that it doesn't break when changing requirements

Reply:
you could perhaps use a regex or some simple logic to grab the lines between two specially formatted comments?

Reply:
I changed it. Now it should work as long as you keep the spaces between the sections

Reply:
and keep the order of the sections. new dependencies can just be added beneath the respective section